### PR TITLE
feat: Introduce FIAT Gateway

### PR DESCRIPTION
**Pull request by juanmahidalgo**
This PR
- Introduces the `FiatGateways` with `WERT` as the first supported type. 
- Moves the `identity` module from the marketplace to here so it's easier to use from the different dapps

### CODE REVIEW

File: src/modules/identity/utilts.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { AuthIdentity } from 'decentraland-crypto-fetch'
+import { localStorageGetIdentity } from '@dcl/single-sign-on-client'
+
+export function* getIdentityOrRedirect(
```
Comment (LautaroPetaccio) [inline src/modules/identity/utilts.ts:4]: It seems that we're using the `getIdentityOrRedirect` from the sagas, should we keep this function as well?

File: src/modules/identity/actions.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { action } from 'typesafe-actions'
+import { AuthIdentity } from '@dcl/crypto'
+
+// Generate identity
+
+export const GENERATE_IDENTITY_REQUEST = '[Request] Generate Identity'
```
Comment (LautaroPetaccio) [inline src/modules/identity/actions.ts:6]: In theory, there's no need to keep this actions, as we won't be handling them anywhere after the refactor in the sites. Are you leaving them to be compatible with what's built today?

File: src/modules/gateway/sagas.ts [hunk 3]
```diff
@@ -83,268 +100,341 @@ export function createGatewaySaga(config: ManaFiatGatewaySagasConfig) {
...
+            )
 
-      if (pendingManaPurchase.gateway === NetworkGatewayType.MOON_PAY) {
-        goToUrl = new MoonPay(config.moonPay).getTransactionReceiptUrl(
-          pendingManaPurchase.id
-        )
+            if (!identity) {
+              yield put(openFiatGatewayWidgetFailure(NO_IDENTITY_ERROR))
+              return
+            } else if (identity) {
```
Comment (LautaroPetaccio) [inline src/modules/gateway/sagas.ts:123]: WDYT about simplifying this?:
```suggestion
            if (!identity) {
              yield put(openFiatGatewayWidgetFailure(NO_IDENTITY_ERROR))
              return
            }
```

File: src/modules/gateway/sagas.ts [hunk 4]
```diff
@@ -83,268 +100,341 @@ export function createGatewaySaga(config: ManaFiatGatewaySagasConfig) {
...
+                        // it's a success event
+                        onSuccess?.({ data: options, type: 'payment-status' })
+                      } else {
+                        onPending?.({ data: options, type: 'payment-status' })
+                      }
+                    }
+                  }
+                })
+
+                wertWidget.open()
```
Comment (LautaroPetaccio) [inline src/modules/gateway/sagas.ts:171]: Would you mind wrapping the this line with a `yield call`? If not, the code will try to instantiate the modal and that might be time consuming.
```suggestion
                yield call([wertWidget, ['open'])
```

**Review by LautaroPetaccio — APPROVED** [review]

File: src/modules/identity/utilts.ts (hunk 1 above)
Comment (juanmahidalgo) [reply src/modules/identity/utilts.ts]: removed!

File: src/modules/identity/actions.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/modules/identity/actions.ts:6]: exactly, leaving it here so it's compatible with how it's used nowadays. If it's not used in the future, I'll remove it.

**Pull request merged**